alg/gdal_interpolateatpoint.cpp

```diff
@@ -158,9 +165,20 @@ template <typename T>
 bool GDALInterpolateAtPointImpl(GDALRasterBand *pBand,
                                 GDALRIOResampleAlg eResampleAlg,
                                 std::unique_ptr<DoublePointsCache> &cache,
-                                const double dfXIn, const double dfYIn, T &out)
+                                double dfXIn, double dfYIn, T &out)
 {
     const gdal::Vector2i rasterSize{pBand->GetXSize(), pBand->GetYSize()};
+
+    if (eResampleAlg == GRIORA_NearestNeighbour)
+    {
+        // Allow input coordinates right at the bottom or right edge
+        // with GRIORA_NearestNeighbour.
+        // "introduce" them in the pixel of the image.
+        if (dfXIn == rasterSize.x())
```

Comment:
```suggestion
        if (dfXIn >= rasterSize.x() && dfXIn <= rasterSize.x()  + 1e-5)
```

Reply:
Should I do this in #12080 ?

Reply:
> Should I do this in #12080 ?

maybe here, if you can push to Dan's branch

Reply:
> > Should I do this in #12080 ?
> 
> maybe here, if you can push to Dan's branch

Ups, I did in https://github.com/OSGeo/gdal/pull/12080 before reading your message.